### PR TITLE
Add custom-role to presence and special view for Recorders

### PR DESCRIPTION
**Pull request by yanas**
Also refactors Toolbar, BottomToolbar and ToolbarToggler.

### CODE REVIEW

File: modules/recorder/Recorder.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+     *
+     * @param conference the {conference} which is to transport
+     * {Recorder}-related information between participants
+     * @param UI the {UI} which is the source (model/state) to be sent to
+     * remote participants if the local participant is the moderator or the
+     * destination (model/state) to receive from the remote moderator if the
+     * local participant is not the moderator
+     */
+    constructor (conference, UI) {
+        var self = this;
```
Comment (hristoterezov) [inline modules/recorder/Recorder.js]: I think this variable is unused.

File: conference.js [hunk 2]
```diff
@@ -721,6 +723,8 @@ export default {
      * Setup interaction between conference and UI.
      */
     _setupListeners () {
+        var self = this;
```
Comment (hristoterezov) [inline conference.js]: I think we don't need that variable. Can't we use "this"?

**Comment by hristoterezov** [comment]
We should split the UI and non-UI part of the recorder class somehow. We can have recorder class in UI folder for the UI part and we can include the rest in conference.js or another file used by conference.js.

**Comment by yanas** [comment]
Thanks @hristoterezov . I'll do that and submit a separate PR for it!

**Pull request merged**